Makefile

```diff
@@ -1,29 +1,82 @@
 SHELL = /bin/bash
-# Variables
-APACHE_ENTRY_PATH := $(shell if [ '$(APACHE_BASE_PATH)' = 'main' ]; then echo ''; else echo /$(APACHE_BASE_PATH); fi)
-KEEP_VERSION ?= 'false'
-LAST_VERSION := $(shell if [ -f '.venv/last-version' ]; then cat .venv/last-version 2> /dev/null; else echo '-none-'; fi)
-VERSION := $(shell if [ '$(KEEP_VERSION)' = 'true' ] && [ '$(LAST_VERSION)' != '-none-' ]; then echo $(LAST_VERSION); else python -c "print __import__('time').strftime('%s')"; fi)
-BRANCH_STAGING := $(shell if [ '$(DEPLOY_TARGET)' = 'dev' ]; then echo 'test'; else echo 'integration'; fi)
-BRANCH_TO_DELETE :=
-CURRENT_DIRECTORY := $(shell pwd)
-DEPLOY_TARGET ?=
+
+# Macro functions
+lastvalue = $(shell if [ -f .venv/last-$1 ]; then cat .venv/last-$1 2> /dev/null; else echo '-none-'; fi;)
+
+define cachelastvariable
+	mkdir -p $(dir $1)
+	test "$2" != "$3" && \
+	    echo "$2" > .venv/last-$4 || :
+endef
+
+# Script variables
 BODID ?=
+DEPLOY_TARGET ?=
+BRANCH_TO_DELETE ?=
 WMSSCALELEGEND ?=
-GIT_BRANCH := $(shell if [ -f '.venv/deployed-git-branch' ]; then cat .venv/deployed-git-branch 2> /dev/null; else git rev-parse --symbolic-full-name --abbrev-ref HEAD; fi)
-HTTP_PROXY := http://ec2-52-28-118-239.eu-central-1.compute.amazonaws.com:80
+
+# Variables
+USER_SOURCE ?= rc_user
+CURRENT_DIRECTORY := $(shell pwd)
+WSGI_APP := $(CURRENT_DIRECTORY)/apache/application.wsgi
 INSTALL_DIRECTORY := .venv
 KML_TEMP_DIR := /var/local/print/kml
 MODWSGI_USER := www-data
+HTTP_PROXY := http://ec2-52-28-118-239.eu-central-1.compute.amazonaws.com:80
+BRANCH_STAGING := $(shell if [ '$(DEPLOY_TARGET)' = 'dev' ]; then echo 'test'; else echo 'integration'; fi)
+GIT_BRANCH := $(shell if [ -f '.venv/deployed-git-branch' ]; \
+							  then cat .venv/deployed-git-branch 2> /dev/null; else git rev-parse --symbolic-full-name --abbrev-ref HEAD; fi)
 NO_TESTS ?= withtests
 NODE_DIRECTORY := node_modules
-PYTHON_FILES := $(shell find chsdi/* -path chsdi/static -prune -o -type f -name "*.py" -print)
+APACHE_ENTRY_PATH := $(shell if [ '$(APACHE_BASE_PATH)' = 'main' ]; then echo ''; else echo /$(APACHE_BASE_PATH); fi)
+DATAGEOADMINHOST ?= data.geo.admin.ch
 SHORTENER_ALLOWED_DOMAINS := admin.ch, swisstopo.ch, bgdi.ch
-SHORTENER_ALLOWED_HOSTS :=
+SHORTENER_ALLOWED_HOSTS ?=
```

Comment:
No fallback value set...?

Reply:
Good question. No we're only using the domains at the moment. Not filtering via host.